src/main/java/org/springframework/integration/aws/dynamodb/BillingMode.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.aws.dynamodb;
+
+/**
+ * The billing mode for DynamoDB tables to be used by the Metadata Store and Registry Lock.
+ * It wraps the values of {@link com.amazonaws.services.dynamodbv2.model.BillingMode} to
+ * support lower case properties as the rest of properties as well as avoid future
+ * billing modes to be available while they are not supported.
+ *
+ * @author Asiel Caballero
+ *
+ * @since 2.3.5
+ *
+ * @see <a href=https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadWriteCapacityMode.html>
+ * Read/Write Capacity Mode</a>
+ */
+public enum BillingMode {
```

Comment:
I don't think we need this class. At least not in this project.
The properties you talk about belongs to Kinesis Binder project. 
In here we just have setters and that's fully OK to expect an enum value from existing `com.amazonaws.services.dynamodbv2.model.BillingMode`.

If you struggle with property names, we can revise it in your Kinesis Binder PR

Reply:
I was trying to replicate the approach used in both `ListenerMode` and `CheckpointMode`. Also to control the values supported by this library. Example, if AWS adds another billing mode in the future you will have a `@since` Javadoc marking the version since this is supported. Do you still prefer to use to pass the Amazon enum instead?

Reply:
True. I do prefer. In fact we just propagate the value to the target API.
In fact without a new introduced value in our API users would need to wait for our new release when just with AWS enum that would have it for free.

The other enums in this project are fully not correlated with AWS API. So, I don't think it is relevant to the subject of this work.

Reply:
Ok, will change.

Reply:
I've address all the comments. Once we get this merged I'll adjust the other PR.